core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java

```diff
@@ -112,71 +98,36 @@ public ByteBuffer read(long offset, long length) throws IOException {
         MetricsSystem.meter(MetricKey.CLIENT_CACHE_BYTES_READ_CACHE.getName()).mark(bytesRead);
         mReadFromLocalCache = true;
       } else {
-        if (!mUfsBlockOptions.isPresent()) {
-          throw new IOException(String.format("Block %d does not have UFS read options, "
-              + "therefore cannot be read from UFS", mBlockMeta.getBlockId()));
+        if (!mUfsBlockReader.isPresent()) {
+          throw new IOException(String.format("Block %d cannot be read from UFS as UFS reader is "
```

Comment:
feel like internalException would be more appropriate

Reply:
done